graphql-java-servlet/src/main/java/graphql/kickstart/servlet/HttpRequestHandlerImpl.java

```diff
@@ -48,37 +51,61 @@ public void handle(HttpServletRequest request, HttpServletResponse response) thr
 
   private void execute(GraphQLInvocationInput invocationInput, HttpServletRequest request,
       HttpServletResponse response) {
-    try {
-      GraphQLQueryResult queryResult = invoke(invocationInput, request, response);
+    if (request.isAsyncSupported()) {
```

Comment:
Unrelated comment. as you are touching this already.
Do you mind changing line 41 to `GraphQLException | IOException e`? `invocationInputParser.getGraphQLInvocationInput` throws IOException (JsonProcessingException or JsonMappingException specifically) in case of unparseable input request. That results in https://github.com/graphql-java-kickstart/graphql-java-servlet/issues/258

Reply:
Are those the only IOException instances that can happen here? Just making sure that we're not also catching others that would qualify as legitimate server errors.

Reply:
I added JsonProcessingException instead of IOException.